File: backend/models/telemetry.go
package models

import "time"

// RoomData represents the unified data structure for a smart home room
type RoomData struct {
	ID                int               `json:"id"`
	Name              map[string]string `json:"name"`
	Temperature       string            `json:"temperature"`
	LastUpdate        time.Time         `json:"last_update"`
	TargetTemperature float64           `json:"target_temperature" bson:"target_temperature"`
	AcStatus          bool              `json:"ac_status"`
}

// BLETelemetryInput — DTO для приема JSON от scripts/ble_bridge.py
type BLETelemetryInput struct {
	DeviceID    string  `json:"device_id"`
	RSSI        int     `json:"rssi"`
	Timestamp   int64   `json:"timestamp"`
	SensorType  string  `json:"sensor_type"`
	Temperature float64 `json:"temperature"`
	Humidity    float64 `json:"humidity"`
	Battery     int     `json:"battery"`
	VoltageMV   int     `json:"voltage_mv"`
}

File: frontend/src/types/telemetry.ts
/**
 * Localized name mapping enforced by the backend translation layers
 */
export interface LocalizedName {
  en: string;
  ru: string;
  et?: string;
  [key: string]: string | undefined;
}

/**
 * Unified data structure representing a smart home room instance.
 * Mirrors the Go backend domains and PostgreSQL / MongoDB combined payload.
 */
export interface RoomData {
  id: number;
  name: LocalizedName;
  temperature: string;
  last_update: string;
  target_temperature: number;
  ac_status: boolean;
}

/**
 * Standard backend response envelope for mutating operations
 */
export interface UpdateResponse {
  message: string;
  room_id: number;
  target_temperature: number;
}
